pkg/kudoctl/cmd/diagnostics/processing_context.go

```diff
@@ -0,0 +1,57 @@
+package diagnostics
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// processingContext - shared data for the resource collectors
+// provides property accessors allowing to define a collector before the data it needs is available
+// provides update callback functions. callbacks panic if called on a wrong type of runtime.object
+type processingContext struct {
+	podNames      []string
+	root          string
+	opName        string
+	opVersionName string
+	instanceName  string
+}
+
+func (ctx *processingContext) attachToRoot() string {
+	return ctx.root
+}
+
+func (ctx *processingContext) attachToOperator() string {
+	return fmt.Sprintf("%s/operator_%s", ctx.root, ctx.opName)
+}
+
+func (ctx *processingContext) attachToInstance() string {
```

Comment:
```suggestion
func (ctx *processingContext) instanceDirectory() string {
```

I think these names describe the function a lot better. The old names may be remnants of the old architecture, but with the new structure it makes it a lot clear:

```
parentDir:      ctx.attachToOperator,
```
feels weird to me, but
```
parentDir:      ctx.operatorDirectory,
```
makes a lot more sense.

Additionally, this func does not attach anything to something, it just returns a directory.